pkg/imgpkg/bundle/images_lock.go

```diff
@@ -13,52 +13,103 @@ import (
 
 //go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/cppforlife/go-cli-ui/ui.UI
 
+type ImageRef struct {
+	lockconfig.ImageRef
+	IsBundle bool
+}
+
+func (i ImageRef) DeepCopy() ImageRef {
+	return ImageRef{
+		ImageRef: i.ImageRef.DeepCopy(),
+		IsBundle: i.IsBundle,
+	}
+}
+
+type ImageRefs struct {
+	refs []ImageRef
+}
+
+func (i *ImageRefs) AddImagesRef(refs ...ImageRef) {
+	for _, ref := range refs {
+		found := false
+		for j, imageRef := range i.refs {
+			if imageRef.Image == ref.Image {
+				found = true
+				i.refs[j] = ref.DeepCopy()
+				break
+			}
+		}
+
+		if !found {
+			i.refs = append(i.refs, ref)
+		}
+	}
+}
+func (i *ImageRefs) Find(ref string) (ImageRef, bool) {
+	for _, imageRef := range i.refs {
+		if imageRef.Image == ref {
+			return imageRef, true
+		}
+	}
+
+	return ImageRef{}, false
+}
+func (i ImageRefs) ImageRefs() []ImageRef {
```

Comment:
if you do not like ImageRefs{}.ImageRefs(), it could be called ImageRefs{}.All()